src/PubSubClient.h

```diff
@@ -34,7 +34,7 @@
  * @brief Maximum packet size defined by MQTT protocol.
  */
 #ifndef MQTT_MAX_POSSIBLE_PACKET_SIZE
-#define MQTT_MAX_POSSIBLE_PACKET_SIZE 268435455
+#define MQTT_MAX_POSSIBLE_PACKET_SIZE ((size_t)268435455)  // might be limited to 65535 if size_t is 16-bit (unsigned int)
```

Comment:
Which platform is 16-bit?
`size_t` is meant to be the size to hold at least a pointer type.

Reply:
If you compile one of the non ESP examples with atmelavr:uno yout get this warning:
```
.pio/libdeps/uno/PubSubClient3/src/PubSubClient.cpp: In member function 'bool PubSubClient::publish(const char*, const char*, uint8_t, bool)':
.pio/libdeps/uno/PubSubClient3/src/PubSubClient.cpp:513:108: warning: large integer implicitly truncated to unsigned type [-Woverflow]
     return publish(topic, (const uint8_t*)payload, payload ? strnlen(payload, MQTT_MAX_POSSIBLE_PACKET_SIZE) : 0, qos, retained);
```
As this is a 16-bit MCU int is here a 16-bit length and the value of `MQTT_MAX_POSSIBLE_PACKET_SIZE` is to big ...